pints/tests/test_residuals_diagnostics.py

```diff
@@ -0,0 +1,199 @@
+#
+# Test the methods in pints.residuals_diagnostics
+#
+# This file is part of PINTS
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing informating, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+import pints
+import pints.residuals_diagnostics
+import pints.toy as toy
+import unittest
+import numpy as np
+import matplotlib
+
+# Select matplotlib backend that doesn't require a screen
+matplotlib.use('Agg')
+
+
+class TestResidualsDiagnostics(unittest.TestCase):
+    """
+    Tests Pints residuals diagnostics methods.
+    """
+    def __init__(self, name):
+        super(TestResidualsDiagnostics, self).__init__(name)
+
+        # Create a single output optimization toy model
```

Comment:
Test initialisation should happen inside a `setupClass` method, see https://docs.python.org/3.5/library/unittest.html#class-and-module-fixtures